src/main/java/bio/terra/workspace/service/controlledresource/model/ControlledResourceMetadata.java

```diff
@@ -0,0 +1,41 @@
+package bio.terra.workspace.service.controlledresource.model;
+
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+import java.util.UUID;
+
+@AutoValue
+public abstract class ControlledResourceMetadata {
+  public abstract UUID workspaceId();
+
+  public abstract UUID resourceId();
+
+  public abstract Optional<String> associatedApp();
+
+  public abstract boolean isVisible();
+
+  public abstract Optional<String> owner();
+
+  public abstract Optional<Object> attributes();
+
+  public static ControlledResourceMetadata.Builder builder() {
+    return new AutoValue_ControlledResourceMetadata.Builder();
+  }
+
+  @AutoValue.Builder
```

Comment:
This is way more work than Immutables.

Reply:
But they both have the same downside: they are not serializable with Jackson without extra Json annotations.

Reply:
They've spent a lot of time getting it tight though. https://immutables.github.io/json.html#jackson. Do POJOs not also need annotations to work with Jackson?